controllers/componentplan_controller.go

```diff
@@ -161,23 +175,36 @@ func (r *ComponentPlanReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 			return ctrl.Result{}, nil
 		}
 
-		_ = r.PatchCondition(ctx, plan, logger, corev1alpha1.ComponentPlanUnInstalling())
+		_ = r.PatchCondition(ctx, plan, logger, revisionNoExist, false, false, corev1alpha1.ComponentPlanUnInstalling())
 		// Perform all operations required before remove the finalizer and allow
 		// the Kubernetes API to remove ComponentPlan
 		go func() {
-			if err = r.doFinalizerOperationsForPlan(ctx, logger, getter, plan); err != nil {
+			r.mu.Lock()
+			if r.inUninstallingMap(plan) {
+				logger.Info("another uninstall running, skip")
+				r.mu.Unlock()
+				return
+			}
+			r.addToUninstallingMap(plan)
+			r.mu.Unlock()
+			err = r.doFinalizerOperationsForPlan(ctx, logger, getter, plan)
+			r.removeFromUninstallingMap(plan)
+			r.removeFromPendingMap(plan)
```

Comment:
why this place contains `removeFromPendingMap` ?

Reply:
when uninstall, helm release will also unintall, So we should reset local cache for this release.